### ISSUE TITLE
server to server sharing from oc8 to oc7 - error: "undefined variable: user"

### ISSUE THREAD

**Issue by derkostka** [owncloud/core#12681]
Steps

Setup OC1 at Version 7.0.2 and OC2 at Daily-Master. 
Add a Folder in OC2 and share it via link,  with password and editable (rw)  for users.
Open the link,  add to owncloud OC1. Add some files into the shared folder from within OC2 and from OC1.

Expected result

File can be downloaded from OC2 via OC1
Files can be added through OC1

Actual result

Both do not work. One can see all files created in OC2, but downloads fail given the error "undefined variable: user"  in sabre dav.  Uploading files to OC2
 over OC1 never finishes uploading. 

Versions

OC1 was 7.0.2
OC2 was pre8

Explanation

I have shared a folder from an OC 8 (yesterdays build) to an OC 7.0.2 instance (web hosted). I can connect those folders and browse it from the OC7 instance, but if i want to download or open a file, i get an error. Here´s the Trace:

```
{"reqId":"6d759195bb3a5079982829f70ef9eff0","remoteAddr":"2001:4ba0:xxxx:139::250:xxxx","app":"webdav","message":"Exception: {\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/dav\\/lib\\/Sabre\\/DAV\\/Server.php(530): OC_Connector_Sabre_File->get()\n#1 \\/var\\/www\\/owncloud\\/lib\\/private\\/connector\\/sabre\\/server.php(89): Sabre\\DAV\\Server->httpGet('etrij.oct2010.0...')\n#2 [internal function]: OC_Connector_Sabre_Server->httpGet('etrij.oct2010.0...')\n#3 \\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/dav\\/lib\\/Sabre\\/DAV\\/Server.php(474): call_user_func(Array, 'etrij.oct2010.0...')\n#4 \\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/dav\\/lib\\/Sabre\\/DAV\\/Server.php(214): Sabre\\DAV\\Server->invokeMethod('GET', 'etrij.oct2010.0...')\n#5 \\/var\\/www\\/owncloud\\/apps\\/files_sharing\\/publicwebdav.php(59): Sabre\\DAV\\Server->exec()\n#6 \\/var\\/www\\/owncloud\\/public.php(48): require_once('\\/var\\/www\\/ownclo...')\n#7 {main}\",\"File\":\"\\/var\\/www\\/owncloud\\/lib\\/private\\/connector\\/sabre\\/file.php\",\"Line\":165}","level":4,"time":"December 03, 2014 23:06:56"}
```
```
{"reqId":"2fa6dcbeeff48ee3ce528758a34bf001","remoteAddr":"2a02:2450:xxxx:1fd:xxxx:xxxx:xxxx:3d02","app":"PHP","message":"Undefined variable: user at \/var\/www\/owncloud\/lib\/private\/connector\/sabre\/principal.php#56","level":3,"time":"December 03, 2014 23:07:01"}
```

**Comment by MorrisJobke** [comment]
Quick fix is merged via #13095

**Comment by MorrisJobke** [comment]
@derkostka Is this issue fixed with your quick fix? Then I would like to close this one

**Comment by derkostka** [comment]
@MorrisJobke The issue is fixed with #13095 , you can close this Issue. Thank you !
